pages/visit/events-exhibitions/index.vue

```diff
@@ -278,130 +255,99 @@ const parsedSeriesAndExhibitions = computed(() => {
-//   const searchAggsResponse = await this.$dataApi.getAggregations(
-//     config.eventsExhibitionsList.filters,
-//     'event'
-//   )
-//   /* console.log(
-//           "Search Aggs Response: " + JSON.stringify(searchAggsResponse)
-//       ) */
-//   searchFilters = [
-//     ...getListingFilters(
-//       searchAggsResponse,
-//       config.eventsExhibitionsList.filters
-//     ),
-//     {
-//       esFieldName: 'past',
-//       inputType: 'single-checkbox',
-//       label: 'Include Past Events',
-//     },
-//   ]
-// }
-
-/* TODO: Enable for search */
-
-// function parseHits(hits = []) {
-//   return hits.value.map((obj) => {
-//     if (obj._source.sectionHandle === 'event') {
-//       return {
-//         ...obj._source,
-//         to: `/${obj._source.to}`,
-//         image: _get(obj._source, 'image[0].image[0]', null),
-//         startDate: _get(
-//           obj._source,
-//           'startDateWithTime',
-//           null
-//         ),
-//         endDate: _get(obj._source, 'endDateWithTime', null),
-//         category: _get(
-//           obj._source,
-//           'eventType[0].title',
-//           null
-//         ),
-//       }
-//     } else if (obj._source.sectionHandle === 'exhibition') {
-//       return {
-//         ...obj._source,
-//         to: `/${obj._source.uri}`,
-//         image: _get(obj._source, 'image[0].image[0]', null),
-//         startDate: _get(obj._source, 'startDate', null),
-//         endDate: _get(obj._source, 'endDate', null),
-//         category: _get(obj._source, 'sectionHandle', null),
-//       }
-//     } else if (obj._source.sectionHandle === 'eventSeries') {
-//       return {
-//         ...obj._source,
-//         to: `/${obj._source.uri}`,
-//         image: _get(obj._source, 'image[0].image[0]', null),
-//         startDate: _get(obj._source, 'startDate', null),
-//         endDate: _get(obj._source, 'endDate', null),
-//         category: 'Event Series',
-//       }
-//     } else {
-//       return {
-//         ...obj._source,
-//         to: `/${obj._source.uri}`,
-//       }
-//     }
-//   })
-// }
-
-// function getSearchData(data) {
-//   // console.log("On the page getsearchdata called " + data)
-//   const filterData =
-//     (data.filters && JSON.stringify(data.filters)) || ''
-//   this.$router.push({
-//     path: '/visit/events-exhibitions',
-//     query: {
-//       q: data.text,
-//       filters: filterData,
-//     },
-//   })
-// }
-
-//
-// fetchOnServer: false,
-// multiple components can return the same `fetchKey` and Nuxt will track them both separately
-// fetchKey: 'events-exhibitions-index',
+onMounted(async () => {
+  await setFilters()
+})
+
+async function setFilters() {
+  const searchAggsResponse = await $dataApi.getAggregations(
+    config.eventsExhibitionsList.filters,
+    'event'
+  )
+  /* console.log(
+          "Search Aggs Response: " + JSON.stringify(searchAggsResponse)
+      ) */
+  searchFilters = [
```

Comment:
replace this with
searchFilters.value = [
     ...getListingFilters(
       searchAggsResponse,
       config.eventsExhibitionsList.filters
     ),
     {
      esFieldName: 'past',
      inputType: 'single-checkbox',
       label: 'Include Past Events',
   },
  ]

Reply:
Since searchFilters is a ref, hence using searchFilters.value

Reply:
Past events single-checkbox filter needs to be added here, sorry for the confusion in the ticket, I thought we had set up past events in Craft. But it is not a filter that comes from ES.